### ISSUE TITLE
 using same dataset and same parameters while trained two different model

### ISSUE THREAD

**Issue by luoguohao** [microsoft/SynapseML#1332]
**Describe the bug**
when i tried to run unit test on linux, `Serialization Fuzzing` test may failed sometimes which shows that the prediction result for the same training dataset  may be inconsistent sometimes  after using deserialized model from disk. so, why it can't be the same ?

**To Reproduce**
i also wrote some codes to test, it may failed sometimes:
```
 test("Verify LightGBM Classifier output same model with same parameters") {
    val fileName = "PimaIndian.csv"
    val labelColumnName = "Diabetes mellitus"
    val fileLocation = DatasetUtils.binaryTrainFile(fileName).toString
    val dataset = readCSV(fileName, fileLocation).repartition(numPartitions)
    val featuresColumn = "_features"
    val rawPredCol = "rawPrediction"
    
    val featurizer = LightGBMUtils.featurizeData(dataset, labelColumnName, featuresColumn)
    val trainData = featurizer.transform(dataset).select(labelColumnName, featuresColumn)
    
    val lgbm1 = new LightGBMClassifier()
      .setLabelCol(labelColumnName)
      .setFeaturesCol(featuresColumn)
      .setRawPredictionCol(rawPredCol)
      .setDefaultListenPort(LightGBMConstants.DefaultLocalListenPort + portIndex)
      .setNumLeaves(5)
      .setNumIterations(10)
      .setObjective(binaryObjective)

    val lgbm2 = new LightGBMClassifier()
      .setLabelCol(labelColumnName)
      .setFeaturesCol(featuresColumn)
      .setRawPredictionCol(rawPredCol)
      .setDefaultListenPort(LightGBMConstants.DefaultLocalListenPort + portIndex)
      .setNumLeaves(5)
      .setNumIterations(10)
      .setObjective(binaryObjective)

    val model1 = lgbm1.fit(trainData)

    println("<=================== model ====================>")
    val model2 = lgbm2.fit(trainData)
    val df1 = model1.transform(trainData)
    val df2 = model2.transform(trainData)
    assertDFEq(df1, df2)
  }
```

**Info (please complete the following information):**
 - SynapseML Version: [v0.18]
 - Spark Version [2.4.3]
 - Spark Platform [Spark On Yarn]

** Stacktrace**
```
Please post the stacktrace here if applicable
```

If the bug pertains to a specific feature please tag the appropriate [CODEOWNER](https://github.com/Microsoft/SynapseML/blob/master/CODEOWNERS) for better visibility

**Additional context**

**Comment by luoguohao** [comment]
@imatiach-msft thanks for the reply, i also tried to using lightGBM distributed case ,it sounds like i can get the same result. i will keep attention on it, thanks~

**Comment by imatiach-msft** [comment]
closing as lightgbm is now deterministic with merged PR https://github.com/microsoft/SynapseML/pull/1387 as long as seed and deterministic=True parameters are set
